cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -412,38 +390,81 @@ def test_cloud_provision_from_template_with_attached_disks(
     if provider.one_of(AzureProvider):
         recursive_update(inst_args, {'environment': {'availability_zone': provisioning("av_set")}})
 
-    device_name = "/dev/sd{}"
+    device_name = "vd{}"
     device_mapping = []
 
-    with provider.mgmt.with_volumes(1, n=disks) as volumes:
-        for i, volume in enumerate(volumes):
-            device_mapping.append((volume, device_name.format(chr(ord("b") + i))))
-        # Set up automate
+    volumes = provider.mgmt.volume_configurations(1, n=disks)
+
+    @request.addfinalizer
+    def delete_volumes():
+        for volume in volumes:
+            provider.mgmt.delete_volume(volume)
+
+    # Set up automate
+    for i, volume in enumerate(volumes, 0):
+        if i == 0:
+            provider.mgmt.capi.volumes.set_bootable(volume, True)
+            device_mapping.append(
+                {'boot_index': 0,
+                'uuid': volume,
+                'device_name': device_name.format(chr(ord("a") + i))})
+        else:
+            device_mapping.append(
+                {'boot_index': -1,
+                'uuid': volume,
+                'device_name': device_name.format(chr(ord("a") + i))})
 
-        method = modified_request_class.methods.instantiate(name="openstack_PreProvision")
+    method = modified_request_class.methods.instantiate(name="openstack_PreProvision")
+
+    view = navigate_to(method, 'Details')
+    former_method_script = view.script.get_value()
+    with update(method):
+        disk_mapping = []
+        for mapping in device_mapping:
+            one_field = dedent("""{{
+                :boot_index => {boot_index},
+                :uuid => "{uuid}",
+                :device_name => "{device_name}",
+                :source_type => "volume",
+                :destination_type => "volume",
+                :volume_size => 1,
+                :delete_on_termination => false
+            }}""")
+            disk_mapping.append(one_field.format(**mapping))
+
+        volume_method = dedent("""
+            clone_options = {{
+            :image_ref => nil,
+            :block_device_mapping_v2 => [
+                {}
+            ]
+            }}
+
+            prov = $evm.root["miq_provision"]
+            prov.set_option(:clone_options, clone_options)
+        """)
+        method.script = volume_method.format(",\n".join(disk_mapping))
```

Comment:
I think this is the only statement that should be in the `with update(method)` context. The for loop and volume_method string above don't modify or depend on the method object.